File: libs/eslint-plugin-east/test-fixtures/proj/bad.ts
import { East, IntegerType, ArrayType, FloatType, variant } from "@elaraai/east";

export const f = East.function([], IntegerType, ($) => {
  const a = $.let([] as number[], ArrayType(FloatType));
  East.value(5n);
  const v = $.const(variant("some", 1n));
  return a.size();
});

File: libs/east-py/packages/east-py/tests/diagnostics/fixtures/prefer-some-none/bad.py
#
# Copyright (c) 2025 Elara AI Pty Ltd
# Licensed under the Business Source License 1.1. See LICENSE.md for details.
#
# ruff: noqa
"""The Option constructors spelled the long way round."""
from east import East, IntegerType, OptionType, variant

EMPTY = variant("none", None, OptionType(IntegerType))  # expect: prefer-some-none
HELD = variant("some", 3, OptionType(IntegerType))  # expect: prefer-some-none


@East.function([IntegerType], OptionType(IntegerType))
def wrapped(b, x):
    return East.if_else(x > 0, variant("some", x, OptionType(IntegerType)), variant("none", None, OptionType(IntegerType)))  # expect: prefer-some-none
